### PR TITLE
Add ability to /play levels by LVL hash

### PR DESCRIPTION
**Pull request by Beyley**

### CODE REVIEW

File: Refresh.GameServer/Endpoints/ApiV3/LevelApiEndpoints.cs [hunk 1]
```diff
@@ -122,7 +122,21 @@ public ApiListResponse<ApiLevelCategoryResponse> GetCategories(RequestContext co
...
+    }
+    
+    [ApiV3Endpoint("levels/hash/{hash}/setAsOverride", HttpMethods.Post)]
+    [DocSummary("Marks the level hash to show in the next slot list gotten from the game")]
+    [DocError(typeof(ApiValidationError), ApiValidationError.HashInvalidErrorWhen)]
+    public ApiOkResponse SetLevelAsOverrideByHash(RequestContext context, GameDatabaseContext database, GameUser user,
+        LevelListOverrideService service, [DocSummary("The hash of level root resource")] string hash)
+    {
+        if (CommonPatterns.Sha1Regex().IsMatch(hash)) 
+            return ApiValidationError.HashInvalidError;
```
Comment (jvyden) [inline Refresh.GameServer/Endpoints/ApiV3/LevelApiEndpoints.cs:137]: :grimacing: 
```suggestion
        if (!CommonPatterns.Sha1Regex().IsMatch(hash)) 
            return ApiValidationError.HashInvalidError;
```

File: Refresh.GameServer/Types/Levels/GameMinimalLevelResponse.cs [hunk 2]
```diff
@@ -48,6 +48,50 @@ public class GameMinimalLevelResponse : IDataConvertableFrom<GameMinimalLevelRes
...
+    public static GameMinimalLevelResponse FromHash(string hash)
+    {
+        return new GameMinimalLevelResponse
+        {
+            LevelId = int.MaxValue,
+            Title = "Hashed Level",
+            IconHash = "0",
+            GameVersion = 0,
+            RootResource = hash,
+            Description = "This is a hashed level, we cant collect any details on it.",
```
Comment (jvyden) [inline Refresh.GameServer/Types/Levels/GameMinimalLevelResponse.cs:65]:
```suggestion
            Title = $"Hashed Level - {hash}",
            IconHash = "0",
            GameVersion = 0,
            RootResource = hash,
            Description = "This is a hashed level. We don't know anything about it.",
```

**Review by jvyden — CHANGES_REQUESTED** [review]

File: Refresh.GameServer/Endpoints/ApiV3/LevelApiEndpoints.cs (hunk 1 above)
Comment (Beyley) [reply Refresh.GameServer/Endpoints/ApiV3/LevelApiEndpoints.cs:137]: ya i should have caught that

**Review by jvyden — APPROVED** [review]
shud be okie

**Pull request merged**